index.js

```diff
@@ -2,14 +2,80 @@ const Vis = require("./graphvis");
 const { Module, render } = require("./visRenderer");
 
 let Sequelize;
-const relationship = ({ source, target, associationType, as }) => {
-  const typeString = {
-    BelongsTo: `[arrowtail=crow, arrowhead=none, dir=both, arrowsize=0.60]`,
-    BelongsToMany: `[arrowtail=crow, arrowhead=crow, dir=both, arrowsize=0.60]`
-  }[associationType];
-  if (typeString) {
-    return `"${source.name}" -> "${target.name}" ${typeString}`;
-  }
+
+const relationships = (associations, arrowShapes = {}, arrowSize = 0.6) => {
+  const mappings = [];
+
+  const mapper = {
+    associationTypes: new Map([
+      ['BelongsToMany', {
+        typeArrowShapes: arrowShapes['BelongsToMany']
+          ? arrowShapes['BelongsToMany']
+          : ['none', 'crow'],
+        items: []
+      }],
+      ['BelongsTo', {
+        typeArrowShapes: arrowShapes['BelongsTo']
+          ? arrowShapes['BelongsTo']
+          : ['crow', 'none'],
+        items: []
+      }],
+      ['HasMany', {
+        typeArrowShapes: arrowShapes['HasMany']
+          ? arrowShapes['HasMany']
+          : ['none', 'crow'],
+        items: []
+      }],
+      ['HasOne', {
+        typeArrowShapes: arrowShapes['HasOne']
+          ? arrowShapes['HasOne']
+          : ['none', 'none'],
+        items: []
+      }],
+    ]),
+  };
+
+  // Separation of association by type to process in the right order
+  associations.forEach(association =>
+    mapper.associationTypes
+      .get(association.associationType)
+      .items.push(association),
+  );
+
+  mapper.associationTypes.forEach(associationType => {
+    associationType.items.forEach(association => {
+      const modelNames = [
+        association.source.name,
+        association.through
+          ? association.through.model.name
+          : association.target.name,
+      ];
+
+      const typeMapping = {
+        [modelNames[0]]: associationType.typeArrowShapes[0],
+        [modelNames[1]]: associationType.typeArrowShapes[1],
+      };
+
+      const thisMapping = mappings.filter(
+        mapping => mapping[modelNames[0]] && mapping[modelNames[1]],
+      );
+
+      if (thisMapping.length === 0) mappings.push(typeMapping);
+      else {
+        thisMapping[0][modelNames[0]] = typeMapping[modelNames[0]];
+        thisMapping[0][modelNames[1]] = typeMapping[modelNames[1]];
+      }
+    });
+  });
+
+  return mappings
+    .map(Object.entries)
+    .map(
+      entries =>
+      entries.length == 1
+      ? `"${entries[0][0]}" -> "${entries[0][0]}" [arrowtail=${entries[0][1]}, arrowhead=none, dir=both, arrowsize=${arrowSize.toString()}]`
```

Comment:
I'm having a hard time following what's going on here. Could we do anything to simplify, or maybe add a comment if it can't be simplified

Reply:
This is probably the most confusing part of the code from PR #7... I tried to quickly modify it to make it a little bit less confusing, but it'll take either a rewrite that'll make the resulting code longer or I can try to add a comment to explain what's happening.

Reply:
I figured this out, and this implementation is VERY clever, so I kept it in and added comments instead! It saves a step, and with the comment, it's understandable, so I'd recommend we keep it!